docs/development.md

```diff
@@ -0,0 +1,102 @@
+**trino-gateway documentation**
+
+<table>
+  <tr>
+    <td><a href="design.md">Design</a></td>
+    <td><b><a href="development.md">Development</a></b></td>
+    <td><a href="security.md">Security</a></td>
+    <td><a href="operation.md">Operation</a></td>
+    <td><a href="gateway-api.md">Gateway API</a></td>
+    <td><a href="resource-groups-api.md">Resource groups API</a></td>
+    <td><a href="routing-rules.md">Routing rules</a></td>
+    <td><a href="references.md">References</a></td>
+  </tr>
+</table>
+
+# Development
+
+## How to setup a dev environment
+
+Step 1: setup mysql. Install docker with docker-compose and run the below
+command when setting up first time:
+
+#### Run the services - mysqldb, two instances of trino
+
+- This setup helps you develop and test any routing rules for the trino
+- Both trino services would have a single `system` catalog
+- Add the catalog properties files in ` bin/localdev/coordinator/` for
+  additional catalogs
+
+```
+cd localdev
+docker-compose up -d
+```
+
+#### Check the "Status' of the services by
+
+`docker-compose ps`
+
+#### Create the schema for the backends, once mysqldb becomes healthy
+
+`docker-compose exec mysqldb sh -c "mysql -uroot -proot123 -hmysqldb -Dtrinogateway < /etc/mysql/gateway-ha-persistence.sql"`
+
+#### Add the backends for mysqldb
+
+`docker-compose exec mysqldb sh -c "mysql -uroot -proot123 -hmysqldb -Dtrinogateway < /etc/mysql/add_backends.sql"`
+
+#### Create the schema for the backends, once postgres becomes healthy
+
+`docker-compose exec postgres sh -c 'PGPASSWORD="P0stG&es" psql -h localhost -p 5432 -U trino_gateway_db_admin -d trino_gateway_db -f /etc/postgresql/gateway-ha-persistence-postgres.sql'`
+
+#### Add the backends for postgres
+
+`docker-compose exec postgres sh -c 'PGPASSWORD="P0stG&es" psql -h localhost -p 5432 -U trino_gateway_db_admin -d trino_gateway_db -f /etc/postgresql/add_backends_postgres.sql'`
+
+It would add 2 trino backend records which can be used for the development and
+testing
+
+### Build and run
+
+Please note these steps have been verified with JDK 8 and 11. Higher versions of
+Java might run into unexpected issues.
+
+run `mvn clean install` to build `trino-gateway`
+
+Edit the [config file](/gateway-ha/gateway-ha-config.yml) and update the mysql
+db information.
+
+```
+cd gateway-ha/target/
+java -jar gateway-ha-{{VERSION}}-jar-with-dependencies.jar server ../gateway-ha-config.yml
+```
```

Comment:
Hi, since update to java 17 causes the following error
> Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make protected final java.lang.Class java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain) throws java.lang.ClassFormatError accessible: module java.base does not "opens java.lang" to unnamed module @49c9930c

How about adding java opts in the doc for now until there is a different shell script or sth that runs the server? I noticed that this option was added when building mvn.
`java --add-opens=java.base/java.lang=ALL-UNNAMED --add-opens=java.base/java.net=ALL-UNNAMED -jar gateway-ha-{{VERSION}}-jar-with-dependencies.jar server ../gateway-ha-config.yml`

Reply:
It would be great if you could send a PR to update the docs for that. I think @willmostly is also looking at working on more the necessary updates towards Java 17 usage.